DiceKeys/Views/Screens/CustomeDiceDemo/CustomDiceFaceView.swift

```diff
@@ -0,0 +1,127 @@
+//
+//  CustomDiceFaceView.swift
+//  DiceKeys
+//
+//  Created by Kevin Shah on 16/01/21.
+//
+
+import SwiftUI
+
+/// DiceFaceManager
+class DiceFaceManager: ObservableObject {
+    
+    @Published var arrDiceFaceModel: [DiceFaceModel] = []
+    
+    var selectedDiceFaceModel: DiceFaceModel? {
+        return arrDiceFaceModel.first { (model) -> Bool in
+            return model.isSelected
+        }
+    }
+    
+    var displayMessage: String {
+        if let model = self.selectedDiceFaceModel {
+            if model.letter == .none {
+                return "One Letter required"
+            } else if model.digit == ._none {
+                return "One Digit required"
+            } else {
+                return "Selected: \(model.letter.rawValue)\(model.digit.rawValue), Orientation"
+            }
+        }
+        return "Please select one square"
+    }
+    
+    func selectNextDiceIfNeeded() {
+        if let selectedModel = self.selectedDiceFaceModel {
+            selectedModel.isSelected = false
+            if let model = arrDiceFaceModel.first(where: { (model) -> Bool in
+                return !model.isDiceFaceModelValid
+            }) {
+                model.isSelected = true
+            } else {
+                if let currentIndex = self.arrDiceFaceModel.firstIndex(where: { (model) -> Bool in
+                    return model.id == selectedModel.id
+                }) {
+                    let nextIndex: Int = currentIndex + 1
+                    if self.arrDiceFaceModel.indices.contains(nextIndex) {
+                        self.arrDiceFaceModel[nextIndex].isSelected = true
+                    }
+                }
+            }
+        }
+    }
+}
+
+/// DiceFaceModel
+class DiceFaceModel: ObservableObject {
+    
+    var id: String = UUID().uuidString
+    
+    @Published var letter: FaceLetter
+    @Published var digit: FaceDigit
+    @Published var orientation: FaceOrientationLetterTrbl = .Top
+    
+    @Published var isSelected: Bool = false
+    
+    var isDiceFaceModelValid: Bool {
+        return (letter != .none) && (digit != ._none)
+    }
+    
+    var faceModel: Face {
+        return Face(letter: self.letter, digit: self.digit, orientationAsLowercaseLetterTrbl: self.orientation)
+    }
+    
+    init(_ letter: FaceLetter, digit: FaceDigit) {
+        self.letter = letter
+        self.digit = digit
+    }
+    
+    init() {
+        self.letter = .none
+        self.digit = ._none
+    }
+}
+
+struct CustomDiceFaceView: View {
```

Comment:
Rename "CustomDiceFaceView" means "EditableDiceKeyView"

As explanation so that you understand the linguistic assumptions behind the change:

1. The view is different from other views because it's editable.
2. Dice is plural, so multiple dice would have "faces" (plural), or DiceFaces
3. But the collection of DiceFaces in this project is a DiceKey, so we can just replace "DiceFaces" with DiceKey